ethcore/src/client/client.rs

```diff
@@ -753,6 +754,42 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
 	fn all_transactions(&self) -> Vec<SignedTransaction> {
 		self.miner.all_transactions()
 	}
+
+	fn take_snapshot(&self, root_dir: &Path) {
+		use pv64::{ManifestData, chunk_blocks, chunk_state};
+
+		let best_header_bytes = self.best_block_header();
+		let best_header = HeaderView::new(&best_header_bytes);
+		let state_root = best_header.state_root();
+
+		trace!(target: "pv64_snapshot", "Taking snapshot starting at block {}", best_header.number());
+
+		let mut path = root_dir.to_owned();
+		path.push("snapshot/");
+		let _ = create_dir(&path);
+
+		// lock the state db while we create the state chunks.
+		let state_hashes = {
+			let state_db = self.state_db.lock().unwrap();
```

Comment:
i had been under the impression that it was necessary for a proper snapshot. our conversation has mitigated these worries.